packages/xsnap/src/replay.js

```diff
@@ -293,17 +296,13 @@ export async function main(argv, { spawn, osType, readdirSync, readFileSync }) {
   await replayXSnap(options, folders, { readdirSync, readFileSync });
 }
 
-/* global require, module, process, require */
-if (typeof require !== 'undefined' && require.main === module) {
+/* global process */
+if (process.argv[1] === new URL(import.meta.url).pathname) {
   main([...process.argv.slice(2)], {
-    // eslint-disable-next-line global-require
-    spawn: require('child_process').spawn,
-    // eslint-disable-next-line global-require
-    osType: require('os').type,
-    // eslint-disable-next-line global-require
-    readdirSync: require('fs').readdirSync,
-    // eslint-disable-next-line global-require
-    readFileSync: require('fs').readFileSync,
+    spawn: childProcessPowers.spawn,
+    osType: osPowers.type,
+    readdirSync: fsPowers.readdirSync,
+    readFileSync: fsPowers.readFileSync,
```

Comment:
Although I found an `if __name__ == '__main__'` idiom for NESM, but even then, I struggled to preserve your intent of limiting the scope of powers to the entrypoint, forcing a certain discipline to thread them through. One of the troubles was having to find names that would not run afoul of eslint name shadowing prohibitions. I arrived at this convention, with `*Powers` module names, which is mildly dissatisfying but got the job done surgically. It might be worthwhile to factor out a `replay-entrypoint.js` that imports this module and the powers instead.

Reply:
GMTA... I did the same thing recently... Modulo Powers vs Top